From foo@baz Tue Apr  9 12:12:43 2002
Date: Mon, 9 Apr 2007 11:52:31 -0400 (EDT)
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: INPUT: remove rwsem use from serio core

The subsystem rwsem is not used by the driver core at all, so the use of
the serio code of it doesn't make any sense.

Acked-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/input/serio/serio.c |    8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

--- a/drivers/input/serio/serio.c
+++ b/drivers/input/serio/serio.c
@@ -119,8 +119,6 @@ static void serio_bind_driver(struct ser
 {
 	int error;
 
-	down_write(&serio_bus.subsys.rwsem);
-
 	if (serio_match_port(drv->id_table, serio)) {
 		serio->dev.driver = &drv->driver;
 		if (serio_connect_driver(serio, drv)) {
@@ -140,27 +138,23 @@ static void serio_bind_driver(struct ser
 		}
 	}
  out:
-	up_write(&serio_bus.subsys.rwsem);
+	return;
 }
 
 static void serio_release_driver(struct serio *serio)
 {
-	down_write(&serio_bus.subsys.rwsem);
 	device_release_driver(&serio->dev);
-	up_write(&serio_bus.subsys.rwsem);
 }
 
 static void serio_find_driver(struct serio *serio)
 {
 	int error;
 
-	down_write(&serio_bus.subsys.rwsem);
 	error = device_attach(&serio->dev);
 	if (error < 0)
 		printk(KERN_WARNING
 			"serio: device_attach() failed for %s (%s), error: %d\n",
 			serio->phys, serio->name, error);
-	up_write(&serio_bus.subsys.rwsem);
 }
 


--- a/drivers/input/gameport/gameport.c
+++ b/drivers/input/gameport/gameport.c
@@ -194,12 +194,10 @@ static void gameport_bind_driver(struct 
 {
 	int error;
 
-	down_write(&gameport_bus.subsys.rwsem);
-
 	gameport->dev.driver = &drv->driver;
 	if (drv->connect(gameport, drv)) {
 		gameport->dev.driver = NULL;
-		goto out;
+		return;
 	}
 
 	error = device_bind_driver(&gameport->dev);
@@ -211,31 +209,23 @@ static void gameport_bind_driver(struct 
 			drv->description, error);
 		drv->disconnect(gameport);
 		gameport->dev.driver = NULL;
-		goto out;
 	}
-
- out:
-	up_write(&gameport_bus.subsys.rwsem);
 }
 
 static void gameport_release_driver(struct gameport *gameport)
 {
-	down_write(&gameport_bus.subsys.rwsem);
 	device_release_driver(&gameport->dev);
-	up_write(&gameport_bus.subsys.rwsem);
 }
 
 static void gameport_find_driver(struct gameport *gameport)
 {
 	int error;
 
-	down_write(&gameport_bus.subsys.rwsem);
 	error = device_attach(&gameport->dev);
 	if (error < 0)
 		printk(KERN_WARNING
 			"gameport: device_attach() failed for %s (%s), error: %d\n",
 			gameport->phys, gameport->name, error);
-	up_write(&gameport_bus.subsys.rwsem);
 }
 
 
